tools/downloader/downloader.py

```diff
@@ -165,12 +185,14 @@ def put(self, hash, path):
 
 def try_retrieve_from_cache(reporter, cache, files):
     try:
-        if all(cache.has(file[0]) for file in files):
-            for hash, destination in files:
+        if all(cache.has(file[0].sha256) for file in files):
+            for model_file, destination in files:
                 reporter.job_context.check_interrupted()
 
                 reporter.print_section_heading('Retrieving {} from the cache', destination)
-                cache.get(hash, destination)
+                if not cache.get(model_file, destination, reporter):
+                    reporter.print('Will retry from the original source.')
```

Comment:
You should add a blank line after this, so that it's visually separate from any messages related to downloading from the source.